reana_server/rest/workspaces.py

```diff
@@ -0,0 +1,72 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of REANA.
+# Copyright (C) 2017, 2018 CERN.
+#
+# REANA is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Reana-Server Workspaces Endpoint."""
+
+import logging
+import traceback
+from flask import Blueprint, jsonify
+
+from reana_commons.config import (
+    ADMIN_ALLOWED_WORKSPACES,
+    SHARED_VOLUME_PATH,
+    DEFAULT_WORKSPACE_PATH,
+)
+
+blueprint = Blueprint("workspaces", __name__)
+
+
+@blueprint.route("/workspaces", methods=["GET"])
+def workspaces():  # noqa
+    r"""Get the list of available workspaces.
+
+    ---
+    get:
+      summary: Get the list of available workspaces.
+      operationId: workspaces
+      description: >-
+        This resource reports the available workspaces in the cluster.
+      produces:
+       - application/json
+      responses:
+        200:
+          description: >-
+            Request succeeded. The response contains the list of all workspaces.
+          schema:
+            type: object
+            properties:
+              workspaces_available:
+                type: array
+                items:
+                  type: string
+          examples:
+            application/json:
+              {
+                "workspaces_available": ["/tmp/reana","/tmp/foo","/var/reana"],
+                "default": "/var/reana"
+              }
+        500:
+          description: >-
+            Request failed. Internal controller error.
+    """
+    try:
+        if ADMIN_ALLOWED_WORKSPACES:
```

Comment:
Please see [my comment](https://github.com/reanahub/reana-commons/pull/286#discussion_r700910771), I think this part could be simplified if we would store the final list in `reana-commons`